js/ball.js

```diff
@@ -22,6 +25,22 @@ class Ball extends p5.Vector {
     this.xVelocity = this.direction.x * speed;
     this.yVelocity = this.direction.y * speed;
   }
+  //change the diection
+  changeDirection(){
+    // TODO:
+    //change direction
+  }
+
+  //constantly checks for collision
+  checkForCollision(x,y){
+    for (var i = 0; i < this.lines.length; i++) {
+      let distance=Math.abs(y-this.lines[i].slope*x-this.lines[i].intercept)/Math.sqrt(1+this.lines[i].slope*this.lines[i].slope)
```

Comment:
Hi can you explain what you are doing here in plain math terms?. 

PS *I am rather new to JS*

Reply:
Hi , thanks for looking into my request! 
So here I am constantly measuring the distance from all the lines (mirror) and the center of the ball . 
For that i am using the formula to get the perpendicular distance from a point to a line .  I needed the equations of lines representing the mirrors and point coordinates of the ball center.

I am able to get the slope and intercept of all the lines representing mirrors  and store it in an array of objects from a function defined in the mirror class .
 when the distance from one of them reaches below a certain value It detects a collision.

Reply:
I also defined a boolean stop in ball class to stop the ball when a collision is detected. We can restart the ball after changing its direction.

Reply:
```suggestion
      let distance=Math.abs(y-this.lines[i].slope*x-this.lines[i].intercept)/Math.sqrt(1+this.lines[i].slope*this.lines[i].slope)
![download](https://user-images.githubusercontent.com/61683657/103412552-a9758f00-4b9b-11eb-95de-4d885f460d16.png)


This is the formula

```

Reply:
So can I get an Update regarding the PR?